menas/src/main/scala/za/co/absa/enceladus/menas/repositories/VersionedMongoRepository.scala

```diff
@@ -131,6 +131,18 @@ abstract class VersionedMongoRepository[C <: VersionedModel](mongoDb: MongoDatab
       set("dateDisabled", ZonedDateTime.now()),
       set("userDisabled", username))).toFuture()
   }
+ def setLockState(name: String, isLocked: Boolean, username: String, datetime: ZonedDateTime = ZonedDateTime.now()): Future[UpdateResult] = {
+    val (dateLocked, userLocked) = if (isLocked) {
+      (datetime, username)
+    } else {
+      (null, null)
+    }
+
+    collection.updateMany(getNameFilter(name), combine(
```

Comment:
I think it updates all versions of the entity. But on second thought, it's probably not a problem. 🤔

Reply:
Yes, I remember this was the purpose